### DISCUSSION TITLE
How to set fixed value to other lightgbm parameters

### DISCUSSION THREAD

**Discussion by keremtatlici** [microsoft/FLAML#271] [Q&A]
I want to set fixed value to other lightgbm parameters that is not on the search tree. For example in [lightgbm doc](https://lightgbm.readthedocs.io/en/latest/Parameters.html#reg_sqrt) there is a parameter named reg_sqrt. I want to make that parameter True while tuning the other parameters.But how to ?

**Comment by qingyun-wu** [comment]
Hi @keremtatlici, you can create a customized LGBM learner and set the parameters you want to fix in the constructor of this customized learner accordingly (similar to the way we set the 'verbose' parameter [here](https://github.com/microsoft/FLAML/blob/6c66cd67f7f3a2762d7ade2b684da6e1ca51606e/flaml/model.py#L285)). Then you can add this customized learner to your automl instance to use it (you can find an example of how to create and add a customized learner in this [notebook](https://github.com/microsoft/FLAML/blob/main/notebook/flaml_lightgbm.ipynb) In [22] `automl.add_learner(learner_name='my_lgbm', learner_class=MyLGBM)`). Let me know if you need further help. Thank you!

**Reply by sonichi** [reply]
@keremtatlici we added a new feature to do this easily. Please check https://microsoft.github.io/FLAML/docs/Use-Cases/Task-Oriented-AutoML#a-shortcut-to-override-the-search-space. Please try version 1.0.4.
